.github/workflows/CI.yml

```diff
@@ -93,8 +93,10 @@ jobs:
     name: Deploy preview
     needs: [eslint, typecheck, setup, validate_generate_files, unit_test]
     if: >-
-      needs.setup.outputs.frontend != 'false' ||
-      needs.setup.outputs.workflows != 'false'
+      (needs.setup.outputs.frontend != 'false' ||
+      needs.setup.outputs.workflows != 'false') &&
+      github.event.pull_request.head.repo.full_name == github.repository &&
+      github.ref_name != 'master'
```

Comment:
What does the `ref_name != master` do - would it still run for PRs towards non-master branches?

Reply:
It skips the deployment on master as we have a separate job for that (that only runs on master).